### PR TITLE
Update README with brief instructions.

### PR DESCRIPTION
**Pull request by varungandhi-src**
Partial progress towards #51.

### Test plan

n/a.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,57 @@
...
+```sh
+npm install -g @sourcegraph/lsif-typescript
+```
+
+### Indexing a TypeScript project
+
+Navigate to the project root, containing `tsconfig.json`.
+
+```sh
+lsif-typescript index
```
Comment (varungandhi-src) [inline README.md:19]: We should have an `npm install` step here to make sure that dependencies are also indexed.

Comment (olafurpg) [reply README.md:19]: I think it's OK to add `yarn install # or npm install` everywhere before we call `lsif-typescript index`. The indexing job succeeds with non-zero exit code if you skip the install step, but the indexed data will have lower quality.

File: README.md [hunk 2]
```diff
@@ -2,4 +2,57 @@
...
+```
+
+### Indexing a JavaScript project
```
Comment (olafurpg) [inline README.md:22]: Might be worth calling out that adding type definition (`@types/*`) devdependencies will improve the quality of the indexed results, even if the codebase itself is pure JS.

File: README.md [hunk 3]
```diff
@@ -2,4 +2,57 @@
...
+```
+
+### Indexing in CI
+
+Add the following run steps to your CI pipeline:
+
+```sh
+npm install -g @sourcegraph/lsif-typescript
+lsif-typescript index
+# From https://github.com/sourcegraph/src-cli/
```
Comment (olafurpg) [inline README.md:51]: Might be worth inlining the curl instructions for Linux here

**Review by olafurpg — APPROVED** [review]
This is a great structure for the readme 👍 A few minor suggestions

**Pull request merged**